fenix/app/src/main/java/org/mozilla/FenixActivity.kt

```diff
@@ -0,0 +1,13 @@
+package org.mozilla
+
+import androidx.navigation.NavDestination
+import androidx.navigation.NavDirections
+import mozilla.components.support.utils.SafeIntent
+import org.mozilla.fenix.BrowserDirection
+
+interface FenixActivity {
+    fun getBreadcrumbMessage(destination: NavDestination): String
+    fun getIntentSource(intent: SafeIntent): String?
+    fun getIntentSessionId(intent: SafeIntent): String?
+    fun getNavDirections(from: BrowserDirection, customTabSessionId: String?): NavDirections?
```

Comment:
I think of these, `getNavDirections` is probably the scariest since it is called from `openToBrowser`. I hadn't considered this, but any reference to `(activity as HomeActivity).openToBrowser` as called throughout the app could imply that the `activity` is actually an `ExternalBrowserActivity`. Have you happened to investigate whether that's the case anywhere?

Reply:
Agreed! Good question, I'm not sure that it _would_ happen right now, but it's possible. 

My preference would be to actually strip out these methods to prevent this, as again it's likely to cause more confusion:

- `openToBrowserAndLoad`
- `openToBrowser`
- `load`
- `handleRequestDesktopMode`

I'll add this as a step in the RFC. I've added a very noddy prototype as a quick demo (commit 3) of my thoughts.